src/compiler/checker.ts

```diff
@@ -13861,7 +13861,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         const typeVariable = getHomomorphicTypeVariable(type);
         if (typeVariable && !type.declaration.nameType) {
             const constraint = getConstraintOfTypeParameter(typeVariable);
-            if (constraint && isArrayType(constraint)) {
+            if (constraint && everyType(constraint, isArrayType)) {

```

Comment:
In a way, this **reverts** the recent change from https://github.com/microsoft/TypeScript/pull/52651 . That issue is now fixed by avoiding getting the apparent type of mapped types within `getApparentTypeOfContextualType` - you can see the change in the commit [here](https://github.com/microsoft/TypeScript/pull/49947/commits/940e1adc260c292f339aae156423a70936c1faad)

I think this is a better fix for that issue anyway, I just didn't figure it out earlier. Getting the "resolved" apparent type for mapped types with tuple constraint (and array) constraints is important for relationship checking. The mapped type that resolves~ to a tuple is usually "normalized" [here](https://github.dev/microsoft/TypeScript/blob/905a0b4e693e8cc13a726ad650b908fee4f75408/src/compiler/checker.ts#L21651), that allows tuple/array-oriented branches to relate things.